pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -539,6 +616,17 @@ func (c *Controller) cleanupEntry(wle config.Config) {
 	log.Infof("cleaned up auto-registered WorkloadEntry %s/%s", wle.Namespace, wle.Name)
 }
 
+func (c *Controller) deleteHealthCondition(wle config.Config) {
+	wle = status.DeleteConfigCondition(wle, status.ConditionHealthy)
```

Comment:
Is deleting it right? Doesn't that mean we mark it as unhealthy?

I would think that we would want to maintain last known state?

Reply:
Yes, my intention is to mark `WorkloadEntry` as unhealthy if sidecar doesn't reconnect within the grace period.

This behaviour is symmetrical to auto-registered `WorkloadEntry`(s) that get deleted if they don't reconnect within the grace period, which results in excluding this workload from load balancing, which is equivalent to marking workload unhealthy. 

I was choosing between:
1. setting `Healthy` condition to `False`
2. setting `Healthy` condition to `Unknown`
3. removing `Healthy` condition

I implemented removing `Healthy` condition since it requires less logic while the effect is the same.